File: src/main/java/com/btr/proxy/selector/pac/Proxy.java
package com.btr.proxy.selector.pac;

public class Proxy {

	public final static Proxy NO_PROXY = new Proxy(null, 0, null);

	public final static String TYPE_HTTP = "http";
	public final static String TYPE_SOCKS4 = "socks4";
	public final static String TYPE_SOCKS5 = "socks5";

	public String type = "http";
	public String host = "";
	public int port = 3128;

	public Proxy(String host, int port, String type) {
		this.host = host;
		this.port = port;
		this.type = type;
	}

}

File: app/src/main/java/com/btr/proxy/selector/pac/Proxy.kt
package com.btr.proxy.selector.pac

class Proxy(
    var host: String? = "",
    var port: Int = 3128,
    var type: String? = TYPE_HTTP
) {
    companion object {
        @JvmField
        val NO_PROXY = Proxy(null, 0, null)

        const val TYPE_HTTP = "http"
        const val TYPE_HTTPS = "https"
        const val TYPE_SOCKS4 = "socks4"
        const val TYPE_SOCKS5 = "socks5"
    }
}
